### ISSUE TITLE
`testability` not returning report fields correctly

### ISSUE THREAD

**Issue by grantfitzsimmons** [specify/specify7#2504]
![chrome_9WhqWnyZWU](https://user-images.githubusercontent.com/37256050/204058327-66fef39e-1cd0-494c-875b-eacd2843841b.gif)

[Associated Query](https://umich-testability.test.specifysystems.org/specify/query/874/) - Returns 71 results correctly on both `edge` and `testability`

![image](https://user-images.githubusercontent.com/37256050/204058264-3f5ed3a1-dc4e-489b-89a6-a0327d522822.png)

The `testability` report should have values for all of those fields.

The report is **herb_all_lbl_all_specify**. Run it with the default query and see the output.

https://umich-testability.test.specifysystems.org/specify/overlay/reports/

Chrome, Windows 11

**Comment by grantfitzsimmons** [comment]
Same issue with the **Gift** report, default report query results.

![image](https://user-images.githubusercontent.com/37256050/204058745-c7c3b3e1-e338-4c03-ae3c-c1dbd1b1e506.png)

**Comment by realVinayak** [comment]
The SpQueryAttributes in testability generate fields with uppercase table names (as @maxxxxxdlp pointed out). Until now, it was not a problem because the backend query runner is not case-sensitive. However, report JasperReports and gift JasperReports definitions are configured to look for lowercase table names (which can be changed in app resources). Thus, the uppercase table names didn't match with JasperReports definitions leading to null values. The uppercase table names are a bug and are easy to fix.
**Fix**:
Change `stringId: [tableList, this.table.name, fieldName].join('.')` to `stringId: [tableList, this.table.name.toLowerCase(), fieldName].join('.')` in toSpQueryAttributes function (https://github.com/specify/specify7/blob/e21fee11e0ba4e1e7cb7a68f312c35a7d291571f/specifyweb/frontend/js_src/lib/components/QueryBuilder/fieldSpec.ts#L123). Doing this fixed the error in my local instance.

**Comment by maxxxxxdlp** [comment]
@realVinayak great job debugging that!

Can you push the fix so that testers can try it out?

Also, can you please add test cases for the `fieldSpec.ts` file? There have been numerous bugs caused by that file, and they take time to debug, so it would be really valuable to have tests for it.

**Comment by grantfitzsimmons** [comment]
<img width="2210" alt="image" src="https://user-images.githubusercontent.com/37256050/204554939-a972bc55-1fec-4b15-ae5b-2bbab806d0f7.png">

Can confirm this bug myself. Good eyes @realVinayak !!!

**Comment by grantfitzsimmons** [comment]
@realVinayak No time to spare! ⏰

**Comment by realVinayak** [comment]
The [e0aa78d](https://github.com/specify/specify7/commit/e0aa78dbed63d19536c3a53ee056b333c1554bdc) would have fixed it, and I checked it and it seems to work. However, some of the reports will not work because I capitalized table names in some reports to check if the capitalization was the problem. That is, all the reports which I capitalized will not work (until I change the table names back to lowercase). For example, https://umich-testability.test.specifysystems.org/specify/resources/app-resource/21/ will not work (since I capitalized them). 

@maxxxxxdlp Is there a way to revert back to the previous reports without manually changing reports?
